backends-common/opensearch/pom.xml

```diff
@@ -71,6 +71,18 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-configuration2</artifactId>
         </dependency>
+        <!-- Needed for opensearch-java dependency -->
+        <dependency>
+            <groupId>org.apache.httpcomponents.client5</groupId>
+            <artifactId>httpclient5</artifactId>
+            <version>5.1.4</version>
+        </dependency>
+        <!-- Needed for opensearch-java dependency -->
+        <dependency>
+            <groupId>org.apache.httpcomponents.core5</groupId>
+            <artifactId>httpcore5</artifactId>
+            <version>5.1.5</version>
+        </dependency>
```

Comment:
Versions are not the same. Is this intendeed?

Reply:
Yes. Just took what was on the java opensearch client side. If you look at the respective versions at maven central as well, you can see those two libs are not strictly coupled:

- https://mvnrepository.com/artifact/org.apache.httpcomponents.client5/httpclient5
- https://mvnrepository.com/artifact/org.apache.httpcomponents.core5/httpcore5

You can see there is no 5.1.5 version for httpclient5, it stops at 1.5.4 (similar with the current 5.2.x upper version actually)

Reply:
httpcore5 is brought by httpclient5 cf https://mvnrepository.com/artifact/org.apache.httpcomponents.client5/httpclient5/5.1.4

IMO we do not need this dependency explicilty

Reply:
Worth a try, forgot which one I added first back then to see the compiler still crying something was missing.

Reply:
I forgot where I found that last time but I remember the scope of those deps in opensearch-java client was runtime, as you can see here: 
[deps.txt](https://github.com/apache/james-project/files/12367658/deps.txt)

Agreed that httpclient5 brings along httpcore5 though, unecessary indeed

Reply:
Conflicts:

```
[INFO] |  +- org.opensearch.client:opensearch-rest-client:jar:2.7.0:runtime
[INFO] |  |  +- (org.apache.httpcomponents:httpclient:jar:4.5.14:runtime - version managed from 4.5.13; omitted for duplicate)
```

Likey an issue... The offender is restassured. Please force this dependency, no need to specify the version as we manage it in the root POM.

Idem forcethe version of httpcore to 4.4.16...

Reply:
Not sure to get it..? httpclient5 and httpcore5 are different libs from httpcore and httpclient? They even have different group ids. I don't see the correlation here

Reminder that the leak blew up in my face big time during perf tests on sandbox, so not only happening in our test suite

Or did i miss something?

Reply:
I could see dependencies between them. Not as unlinked as one might think...